src/yb/yql/pgwrapper/pg_libpq-test.cc

```diff
@@ -190,6 +190,36 @@ TEST_F(PgLibPqTest, Simple) {
   ASSERT_EQ(row, (decltype(row){1, "hello"}));
 }
 
+TEST_F(PgLibPqTest, PgStatIdxScanNoIncrementOnErrorTest) {
+  auto conn = ASSERT_RESULT(Connect());
+
+  std::string createTableStmt = "create table many (";
+  for (int i = 1; i <= 30; ++i)
+    createTableStmt += Format("c$0 int,", i);
+  createTableStmt += "k int primary key)";
+
+  std::string createIndexStmt = "create index on many (";
+  for (int i = 1; i <= 30; ++i)
+    createIndexStmt += Format("c$0,", i);
+  createIndexStmt += "c1 asc)";
+
+  std::string predicateErrorQuery = "select * from many where ";
+  for (int i = 1; i <= 30; ++i)
+    predicateErrorQuery += Format("c$0 = 1 and c$0 < 2 and c$0 > 0 and ", i);
+  predicateErrorQuery += " true";
+
+  ASSERT_OK(conn.Execute(createTableStmt));
+  ASSERT_OK(conn.Execute(createIndexStmt));
+
+  ASSERT_NOK(conn.Fetch(predicateErrorQuery));
+  const auto noIncrementRows = ASSERT_RESULT((conn.FetchRows<int64_t>("select idx_scan from pg_stat_user_indexes")));
+  ASSERT_EQ(noIncrementRows, (decltype(noIncrementRows){0, 0}));
+
+  ASSERT_TRUE(ASSERT_RESULT(conn.HasIndexScan("select * from many where c1 = 1")));
+  const auto incrementRows = ASSERT_RESULT((conn.FetchRows<int64_t>("select idx_scan from pg_stat_user_indexes")));
+  ASSERT_EQ(incrementRows, (decltype(incrementRows){0, 0})); // should expect 0, 1
```

Comment:
@jasonyb any ideas why the count isn't incremented here when an index scan happens successfully? It gives
```
                          indexrelname                           | idx_scan
-----------------------------------------------------------------+----------
 many_pkey                                                       |        0
 many_c1_c2_c3_c4_c5_c6_c7_c8_c9_c10_c11_c12_c13_c14_c15_c16_idx |        1
 ```
 on my local, but test results seem to give 0, 0.

Reply:
`conn.HasIndexScan` doesn't run the query

Reply:
I've also tried the following, but these attempts also don't increase the count
```
ASSERT_OK(conn.Fetch("select * from many where c1 = 1"));
 ```
 and
 ```
 auto values = ASSERT_RESULT(conn.FetchRows<int32_t>("select c1 from many where c1 = 1"));
 ASSERT_EQ(values.size(), 0);
 ```

Reply:
This was nontrivial information: stats are updated only periodically.  I have documented/updated the test myself.